src/Symfony/Component/Validator/Constraints/File.php

```diff
@@ -88,23 +88,18 @@ public function __get($option)
 
     private function normalizeBinaryFormat($maxSize)
     {
-        $sizeInt = (int) $maxSize;
-
+        $factors = array(
+                'k' => 1000,
+                'ki' => 1 << 10,
+                'm' => 1000000,
+                'mi' => 1 << 20,
+        );
         if (ctype_digit((string) $maxSize)) {
-            $this->maxSize = $sizeInt;
-            $this->binaryFormat = null === $this->binaryFormat ? false : $this->binaryFormat;
-        } elseif (preg_match('/^\d++k$/i', $maxSize)) {
-            $this->maxSize = $sizeInt * 1000;
-            $this->binaryFormat = null === $this->binaryFormat ? false : $this->binaryFormat;
-        } elseif (preg_match('/^\d++M$/i', $maxSize)) {
-            $this->maxSize = $sizeInt * 1000000;
+            $this->maxSize = (int) $maxSize;
             $this->binaryFormat = null === $this->binaryFormat ? false : $this->binaryFormat;
-        } elseif (preg_match('/^\d++Ki$/i', $maxSize)) {
-            $this->maxSize = $sizeInt << 10;
-            $this->binaryFormat = null === $this->binaryFormat ? true : $this->binaryFormat;
-        } elseif (preg_match('/^\d++Mi$/i', $maxSize)) {
-            $this->maxSize = $sizeInt << 20;
-            $this->binaryFormat = null === $this->binaryFormat ? true : $this->binaryFormat;
+        } elseif (preg_match('/^(\d++)('.implode('|', array_keys($factors)).')$/i', $maxSize, $matches)) {
+            $this->maxSize = $matches[1] * $factors[$unit = strtolower($matches[2])];
```

Comment:
The `strtolower` call is necessary because of the `i` modifier in the regex, which effectively allows any case for the unit (even `mI`). The `$unit` variable can be avoided, I'll remove it rightaway.